pkg/gcp/build/build.go

```diff
@@ -56,6 +59,49 @@ type Options struct {
 	EnvPassthrough string
 }
 
+func PrepareBuilds(o *Options) error {
+	if o.ConfigDir == "" {
+		return errors.New("expected a config directory to be provided")
+	}
+
+	if bazelWorkspace := os.Getenv("BUILD_WORKSPACE_DIRECTORY"); bazelWorkspace != "" {
+		if err := os.Chdir(bazelWorkspace); err != nil {
+			return errors.Wrapf(err, "failed to chdir to bazel workspace (%s)", bazelWorkspace)
+		}
+	}
+
+	if o.BuildDir == "" {
+		o.BuildDir = o.ConfigDir
+	}
+
+	logrus.Infof("Build directory: %s\n", o.BuildDir)
+
+	// Canonicalize the config directory to be an absolute path.
+	// As we're about to cd into the build directory, we need a consistent way to reference the config files
+	// when the config directory is not the same as the build directory.
+	absConfigDir, absErr := filepath.Abs(o.ConfigDir)
+	if absErr != nil {
+		return errors.Wrapf(absErr, "could not resolve absolute path for config directory")
+	}
+
+	o.ConfigDir = absConfigDir
+	o.CloudbuildFile = path.Join(o.ConfigDir, o.CloudbuildFile)
+
+	configDirErr := o.ValidateConfigDir()
+	if configDirErr != nil {
+		return errors.Wrapf(configDirErr, "could not validate config directory")
+	}
```

Comment:
Why not put it in one if statement?

Reply:
I have a slight readability preference with the current form.